MIP/mip-58/README.md

```diff
@@ -0,0 +1,462 @@
+7. **Infrastructure Simplification**:
+   - Infrastructure is still incomplete for the current HTLC bridge design and we are struggling to find sound solutions that balance UI/UX and security. For example, [this issue](https://github.com/movementlabsxyz/movement/issues/838) regarding UX vs number of confirmations to require on the Ethereum side, is still under debate.
+   - Because of the over-engineered design, infrastructure is prone to error and we might end up being damaged by the amount of infrastructure we have not built yet and has to be built for the relayer to fully function.
+   - We could strip down the Relayer code and achieve a final design much more quickly.
+
+The proposed two-transaction design mitigates these issues, creating a safer, faster, and user-friendly bridge while maintaining operational reliability.
+
+## Specification
+
+The simplified bridge design focuses on minimizing complexity and maximizing security:
+
+### Protocol Description
+
+The bridge is a simpler version of the current implementation, which relies on the assumption that the Relayer Keys are not to be compromised.
+
+L1 -> L2
+
+1. User initiates a bridge transfer on L1. Contract stores a mapping of the user bridgeTransferIds for easy access. It transfers from the user MOVE amount to the contract. Transaction emits originator, recipient, amount and nonce.
+2. Relayer awaits for finalization of the transaction on L1.
+3. Relayer completes the bridge on L2.
+4. Completion transaction verifies that the transaction is truthful by comparing the provided bridgeTransferId hash and the emitted values of originator, recipient, amount and nonce. Finally, it mints MOVE to the recipient address.
+5. User is notified on the frontend that their transaction has been completed.
+6. In case of a rare case of a reorg on the recipient network that leads to the transaction removal from the user perspective, the transaction can be redone and relayer multisig can handle it.
+
+![L1-L2](l1l2.png)
+
+L2 -> L1
+
+1. User initiates a bridge transfer on L2. Contract stores a mapping of the user bridgeTransferIds for easy access. It transfers from the user MOVE amount to the contract and burns it. Transaction emits originator, recipient, amount and nonce.
+2. Relayer awaits for finalization of the transaction on L2.
+3. Relayer completes the bridge on L1.
+4. Completion transaction verifies that the transaction is truthful by comparing the provided bridgeTransferId hash and the emitted values of originator, recipient, amount and nonce. Finally, it transfers MOVE to the recipient address, which has been previously bridged from L1.
+5. User is notified on the frontend that their transaction has been completed.
+6. In case of a rare caase of a reorg on the recipient network that leads to the transaction removal from the user perspective, the transaction can be redone and relayer multisig can handle it.
```

Comment:
The problem is not a re-org on the target chain, but on the source chain. 
This is explained in the [issue 838](https://github.com/movementlabsxyz/movement/issues/838).

Say if a user is granted k MOVE on L2 on the counterparty via Tx t1, but we discover that the initial transfer on L1 was reverted, how do we undo transactions that have been processed on P2 and depend on t1?
It is not a problem of permission ("the relayer can handle it", that's very hand-wavy), but a problem of undoing tx that could in several different blocks, could impact a multitude of users etc

Reply:
Both reorgs are issues but they are different issues. Just highlighted that in the case of a bug, we guarantee that liveness of Relayer will not impede completion by the maintainer party. In case of a collision a double spend is impossible.